### PR TITLE
add search trade for timeless jewels

### LINKED ISSUE #5401: Add Trade Search finder for Timeless Jewels
### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### Is your feature request related to a problem?

I'd like to be able to search the trade website easily to find the resultant found jewels in Timeless Finder.

### Describe the solution you'd like

After finding the results that match the Timeless Jewel Finder filter I would like to be able to easily access the trade website to find jewels from there.

### Describe alternatives you've considered

Currently, there is no way to do this besides manually creating your own search.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by meepen**
Fixes #5401.

### Description of the problem being solved:
Currently, there is no easy way to access the Trade website to find the Timeless Jewels from the resultant list in the Timeless Jewel Finder. This PR creates a button to search the website for jewels in bulk.

### Steps taken to verify a working solution:
- I have interacted with the Timeless Jewel Finder
- I have tested this in a working build.
- I have accessed the URLs created by the Searcher to ensure they are correct under Rakiata

### Before screenshot:
image

### After screenshot:
image

### CODE REVIEW

**Comment by meepen** [comment]
Currently need help working out how to get or create a dropdown for leagues.

**Comment by QuickStick123** [comment]
It will probably be easy to interact with the trader as this has all this functionality with leagues and realms (in an open PR). Feel free to message one of our maintainers on discord

> There is a [Discord](https://discordapp.com/) server for active development on the fork and members are happy to answer your questions there. If you are interested in joining, send a private message to any of Cinnabarit#1341, LocalIdentity#9871, Yamin#5575 and we'll send you an invitation.

**Comment by QuickStick123** [comment]
Fixes https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/5013 as well

**Comment by meepen** [comment]
Should be good all alone and not use POESESSID anymore

**Comment by QuickStick123** [comment]
Overall this feature is excellent and behaves as expected. Things that could be improved: (I feel people might ask for)
- User settable number of results to add e.g. top 20. As people might only want to look for the best 10 jewels for there build this would let them do this to create a live search for them more easily. Rather than the default 50 you have currently
- Some way to go back to previous search copies e.g. I have clicked the button twice but I would like to copy page 1 again. I then currently have to close timeless window, generate weights and search again before I can research these.

**Comment by meepen** [comment]
Ideally I would like to select (with clicking first element and shift clicking second) and have it allow you to generate a link. I would also like to have a page dropdown which would highlight which ones are being searched per page.

**Comment by meepen** [comment]
But I have no idea how to do any of that with your API :( I can try a bit longer but I have work during the week so I might not have the most time.

**Comment by QuickStick123** [comment]
You would implement it by using IsKeyDown("CTRL") and the existing TimelessJewelListControlClass:OnSelClick

**Comment by meepen** [comment]
I've added a (probably newbie) way to select multiple values and defaulted it to select the first value with 10 rows, and after clicking copy trade url it will proceed to the next batch after one is already highlighted.

Path{space}of{space}Building_gezWR0CJEL

**Comment by meepen** [comment]
Note you can select the previous ones again by just clicking the other rows.

**Comment by QuickStick123** [comment]
This looks pretty excellent maybe a special colour for the first and last? So if it spans more than one screen you can tell it is not at the start.

**Comment by meepen** [comment]
Done

**Review by Wires77 — COMMENTED** [review]
Haven't reviewed this in full yet, but looks alright from what I see.  I'm unsure about the UX of clicking the copy button again advancing to the next section of jewels, however.  I don't think the average user will think to do that/expect that behavior.

**Comment by meepen** [comment]
Fixed some issues and I agree and made the label change after first click.
Path{space}of{space}Building_jDM07py1sn

**Comment by Crazy-KSK** [comment]
This might be too much to ask but a toggle to search for all types/ignore keystone/conqueror, so as to not have to make 3 separate searches, would be very welcome.
Thanks.

**Comment by meepen** [comment]
it might be best to save that for another pr which could possibly properly sort by dps.

**Comment by KonstantinosKoubetsos** [comment]
This is actually sick, can't wait for approval. The hours wasted copy pasting different timeless trade search configs for different builds..

File: src/Classes/ListControl.lua
Comment (Wires77) [inline src/Classes/ListControl.lua:261]:
```suggestion
			if not self.SetHighlightColor or not self:SetHighlightColor(index, value) then
```

File: src/Classes/ListControl.lua
Comment (Wires77) [inline src/Classes/ListControl.lua:345]:
```suggestion
		if self:SelectIndex(newSelect) then
```

File: src/Classes/TimelessJewelListControl.lua
Comment (Wires77) [inline src/Classes/TimelessJewelListControl.lua:45]:
```suggestion
	if IsKeyDown("SHIFT") and self.selIndex then
```

**Review by Wires77 — APPROVED** [review]
Seems to work pretty well from my testing, nice job.

**Pull request merged**
